robottelo/issue_handlers/bugzilla.py

```diff
@@ -232,6 +233,28 @@ def get_data_bz(bz_numbers, cached_data=None):  # pragma: no cover
             if 'data' in item
         ]
 
+    # Ensure API key is set
+    if not settings.bugzilla.api_key:
+        LOGGER.warning(
+            "Config file is missing bugzilla api_key "
+            "so all tests with skip_if_open mark is skipped. "
+            "Provide api_key or a bz_cache.json."
+        )
+        # Provide default data for collected BZs
+        return [
+            {
+                "id": number,
+                "is_open": True,  # All marked is skipped
+                "is_deselected": False,  # nothing is deselected
+                "status": "",
+                "resolution": "",
```

Comment:
I have a query here , in return why `status, resolution, cf_clone_of, clone_ids` are required without values ? or am i missing something here ?

Reply:
You are right @mirekdlugosz to skip or not so skip will be decided by 
`is_open": True` and `"is_deselected": False`. But there this will also be looked up id BZ has duplicates ot clones, so we add that there just to defult it as a NO duplicate and having no clones.

Reply:
in the future, doing dict lookups with .get() avoids this :)

Reply:
@JacobCallahan I am not a big fan of using `.get` for some cases,  that data is schemaless yet, so we need it to not silence the problem if for example Bugzilla API changes in future.

I would prefer to make some kind of Type Serialization, define a `Bug` Schema using Marshmallow and then serialize API response to that object.

Reply:
@JacobCallahan not Marshmallow, I mean `Serde` (inspired by Rust) https://pypi.org/project/serde/